File: duplicate-assert/java-junit/TestLiteWorkflowAppParser_testParserGlobalExtensionActionsNoGlobal.java
public class TestLiteWorkflowAppParser extends XTestCase {
    public void testParserGlobalExtensionActionsNoGlobal() throws Exception {
        LiteWorkflowAppParser parser = new LiteWorkflowAppParser(null,
            LiteWorkflowStoreService.LiteControlNodeHandler.class,
            LiteWorkflowStoreService.LiteDecisionHandler.class,
            LiteWorkflowStoreService.LiteActionHandler.class);

        // If no global section is defined, some extension actions (e.g. hive) must still have name-node and job-tracker elements
        // or the handleGlobal() method will throw an exception

        parser.validateAndParse(IOUtils.getResourceAsReader("wf-schema-valid-global-ext-no-global.xml", -1), new Configuration());

        try {
            parser.validateAndParse(IOUtils.getResourceAsReader("wf-schema-invalid-global-ext-no-global.xml", -1),
                    new Configuration());
            fail();
        }
        catch (WorkflowException ex) {
            assertEquals(ErrorCode.E0701, ex.getErrorCode());
        }
        catch (Exception ex) {
            fail();
        }
    }
}

File: duplicate-assert/CSharpXUnit/CSharpXUnit.Tests/Tests/TestLiteWorkflowAppParser_testParserGlobalExtensionActionsNoGlobal.cs
public class TestLiteWorkflowAppParser
{
    [Fact]
    public void TestParserGlobalExtensionActionsNoGlobal()
    {
        var parser = new LiteWorkflowAppParser(null,
            typeof(LiteWorkflowStoreService.LiteControlNodeHandler),
            typeof(LiteWorkflowStoreService.LiteDecisionHandler),
            typeof(LiteWorkflowStoreService.LiteActionHandler));

        parser.ValidateAndParse(IOUtils.GetResourceAsReader("wf-schema-valid-global-ext-no-global.xml", -1), new Configuration());

        Exception exception = Record.Exception(() =>
        {
            parser.ValidateAndParse(IOUtils.GetResourceAsReader("wf-schema-invalid-global-ext-no-global.xml", -1),
                new Configuration());
        });

        Assert.IsType<WorkflowException>(exception);
        Assert.Equal(ErrorCode.E0701, ((WorkflowException)exception).ErrorCode);
    }
}
